Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -0,0 +1,373 @@
+        // Give it time to collect a profile, otherwise there will be no samples.
+        DispatchQueue.global().asyncAfter(deadline: .now() + duration) {
+            span.finish()
+
+            guard let envelope = self.fixture.client.captureEnvelopeInvocations.first else {
+                XCTFail("Expected to capture at least 1 event")
+                return
+            }
+            XCTAssertEqual(1, envelope.items.count)
+            guard let profileItem = envelope.items.first else {
+                XCTFail("Expected at least 1 additional envelope item")
+                return
+            }
+            XCTAssertEqual("profile", profileItem.header.type)
+            self.assertValidProfileData(data: profileItem.data, transactionEnvironment: transactionEnvironment, duration: duration, numberOfTransactions: numberOfTransactions, customAssertions: customAssertions)
+            profileExpectation.fulfill()
+        }
+
+        // Some busy work to try and get it to show up in the profile.
+        let str = "a"
+        var concatStr = ""
+        for _ in 0..<100_000 {
+            concatStr = concatStr.appending(str)
+        }
+
+        waitForExpectations(timeout: duration + 3.0) {
+            if let error = $0 {
+                print(error)
+            }
+        }
+    }
+
+    func assertValidProfileData(data: Data, transactionEnvironment: String = kSentryDefaultEnvironment, duration: TimeInterval = 2.0, numberOfTransactions: Int = 1, customAssertions: (([String: Any]) -> Void)? = nil) {
+        let profile = try! JSONSerialization.jsonObject(with: data) as! [String: Any]
+        XCTAssertEqual("Apple", profile["device_manufacturer"] as! String)
+        XCTAssertEqual("cocoa", profile["platform"] as! String)
+        XCTAssertNotNil(profile["transactions"])
+        if let transactions = profile["transactions"] as? [[String: String]] {
+            XCTAssertEqual(transactions.count, numberOfTransactions)
+            for transaction in transactions {
+                XCTAssertEqual(fixture.transactionName, transaction["name"])
+                XCTAssertNotNil(transaction["id"])
+                if let idString = transaction["id"] {
+                    XCTAssertNotEqual(SentryId.empty, SentryId(uuidString: idString))
+                }
+                XCTAssertNotNil(transaction["trace_id"])
+                if let traceIDString = transaction["trace_id"] {
+                    XCTAssertNotEqual(SentryId.empty, SentryId(uuidString: traceIDString))
+                }
+                XCTAssertEqual(transactionEnvironment, transaction["environment"])
+                XCTAssertNotNil(transaction["trace_id"])
+                XCTAssertNotNil(transaction["relative_start_ns"])
+                XCTAssertNotNil(transaction["relative_end_ns"])
+            }
+        } else {
+            XCTFail("Transaction information in profile payload not of expected type.")
+        }
```

Comment:
These are the only new assertions added to this function, now that we are sending some transaction info in the profiling payload.